OMCompiler/Compiler/runtime/systemimpl.c

```diff
@@ -306,7 +306,8 @@ extern char* SystemImpl__pwd(void)
     return NULL;
   }
 
-  WCHAR unicodePath[bufLen];
+  // WCHAR unicodePath[bufLen];
+  WCHAR* unicodePath = (WCHAR*)omc_alloc_interface.malloc(bufLen * sizeof(WCHAR));
   if (!GetCurrentDirectoryW(bufLen, unicodePath)) {
     c_add_message(NULL,-1,ErrorType_scripting,ErrorLevel_error,gettext("GetCurrentDirectoryW failed."),NULL,0);
```

Comment:
You're leaking memory here. That's why you use VLAs... Or maybe not, if it's GC'ed, but then why not use malloc_atomic?

Reply:
> You're leaking memory here. 

There is a `GC_free` a few lines below it. Anyway is it not supposed to be freed automatically? Is that not the point of the GC. In fact IIRC it is recommended not to free small chunks manually. Are we expecting the runtime to switch to the memory pool in the future?    

> That's why you use VLAs...Or maybe not

Not. That does not make much sense. These are two different concepts.

> but then why not use malloc_atomic?

You are right. It is probably better since it is not made up of pointers. It is changed now.